core/types/deposit_tx.go

```diff
@@ -22,6 +22,10 @@ import (
 	"github.com/ethereum/go-ethereum/common"
 )
 
+// L1InfoDepositerAddress is the only "from" address authorized to send the L1 Attributes Tx
+// It does not correspond to a known EOA. It is not imported from the `op-node` to avoid a circular dependency.
+var L1InfoDepositerAddress = common.HexToAddress("0xdeaddeaddeaddeaddeaddeaddeaddeaddead0001")
+
```

Comment:
Why not add the systemTx bool to the deposit tx type? The field doesn't cost us much (disk space, but no L1 data) and we could set it in the rollup node that creates the deposit transaction. That way we don't have to hardcode special addresses, and have more flexibility in future upgrades (e.g. only updating the rollup node).

Reply:
This was a more isolated change, but yea, adding the bool the deposit tx type also works & gives more flexibility to the rollup node to designate system transactions

Reply:
@trianglesphere can you add the bool to the tx content?

Reply:
Sure. I'll get this PR updated with it & then do all the updates in the rollup node.

Reply:
There are a bunch of other already approved fixes in PRs. After you update this PR I can cherry-pick the commits onto the history branch, rebase onto latest upstream geth, squash, and make a PR to monorepo. Easier to do this once than 4x.